File: app/src/main/java/com/app/cheffyuser/home/fragments/PlateReviewFragment.java
package com.app.cheffyuser.home.fragments;


import android.content.Intent;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.cardview.widget.CardView;
import androidx.fragment.app.Fragment;

import com.app.cheffyuser.R;
import com.app.cheffyuser.home.activities.FoodAddToCartActivity;

/**
 * A simple {@link Fragment} subclass.
 */
public class PlateReviewFragment extends Fragment {


    public PlateReviewFragment() {
        // Required empty public constructor
    }


    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View view= inflater.inflate(R.layout.fragment_plate_review, container, false);

        CardView cartButton=view.findViewById(R.id.cardbutton);

        cartButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent=new Intent(getActivity(), FoodAddToCartActivity.class);
                startActivity(intent);
            }
        });

        return view;
    }

}

File: app/src/main/java/com/app/cheffyuser/home/fragments/PlateReviewFragment.kt
package com.app.cheffyuser.home.fragments


import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import androidx.fragment.app.Fragment
import androidx.lifecycle.ViewModelProviders

import com.app.cheffyuser.R
import com.app.cheffyuser.home.adapter.ReviewsAdapter
import com.app.cheffyuser.home.viewmodel.HomeViewModel
import com.app.cheffyuser.utils.hideView
import com.app.cheffyuser.utils.showView
import kotlinx.android.synthetic.main.fragment_plate_review.*
import kotlinx.android.synthetic.main.no_item_layout.*


/**
 * A simple [Fragment] subclass.
 */
class PlateReviewFragment : BaseFragment() {

    private val vm: HomeViewModel by lazy {
        ViewModelProviders.of(getActivity()!!).get(HomeViewModel::class.java)
    }

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_plate_review, container, false)
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)

        no_item_text.text = "No reviews yet, be the first"


        reviewlist.setHasFixedSize(true)

        val reviews = vm.platesResponse.value?.reviews

        if (reviews.isNullOrEmpty()) {

            noitem_layout.showView()
            reviewlist.hideView()

        } else {
            reviewlist.showView()
            noitem_layout.hideView()

            val adapter = ReviewsAdapter(context!!, reviews.toMutableList())
            reviewlist.adapter = adapter

        }
    }

}
